rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -1437,11 +1442,31 @@ impl MergeInsertJob {
 
         let has_scalar_index = self.join_key_as_scalar_index().await?.is_some();
 
+        // Check if this is a delete-only operation (no update/insert writes needed from source)
+        // For delete-only, we don't need the full source schema, just key columns for matching
+        let is_delete_only = matches!(
```

Comment:
this name seems misleading, since it can be both delete and do nothing, it's more like `no_upsert`?